File: geeksforgeeks/Algorithms/Searching and Sorting/insertion_sort.java
import java.util.*;
public class Main {
	static void InsertionSort(int arr[],int n) //Insertion Sort Function
	{
		for(int i=0;i<n;i++)
		{
			int m=i;
			for(int j=i-1;j>=0;j--)
			{
				if(arr[j]>arr[m])
				{
					int temp=arr[j];
					arr[j]=arr[m];
					arr[m]=temp;
					m--;
				}
			}
		}
	}
	public static void main(String[] args) 
	{
		int arr[]={12,11,5,3,6,343,2,43,1,232,45};
		int n = arr.length;
		
		InsertionSort(arr,n);
		
		for(int i=0;i<n;i++)
		{
			System.out.println(arr[i]);
		}
		
	}  
}

File: geeksforgeeks/Top 10 Algorithms and Data Structures/Searching and Sorting/Insertion Sort.cpp
#include <iostream>
using namespace std;

// Function for Insertion Sort
void insertion_sort(int arr[], int n)
{
	int j,temp;
	for(int i = 0;i < n;i++)
	{
		j = i;
		while(j > 0 && arr[j] < arr[j-1])
		{
			temp = arr[j];
			arr[j] = arr[j-1];
			arr[j-1] = temp;
			j--;
		}
	}
}

int main() {
	// Test Array
    int arr[] = {7,2,5,2,4,1,6,112,43,42,32121,32,1,43,2};
    int n = sizeof(arr)/sizeof(arr[0]);
    
    insertion_sort(arr,n);
    
    // Sorted Output
    for(int i = 0;i < n;i++)
    {
    	cout << arr[i] << " ";
    }
    return 0;
}